src/formatters/json/mod.rs

```diff
@@ -1,10 +1,69 @@
+use std::borrow::Cow;
+use std::marker::PhantomData;
 use std::io::Write;
 use std::num::Wrapping;
-use events::Event;
+use events::{Event, Value};
+use super::{ValueFormatter, ValueFormatterVisitor, format_vec};
 use std::error::Error;
-use serde_json;
 use LogLevel;
 
+#[derive(Default)]
+#[doc(hidden)]
+pub struct JsonValueFormatter<'a> {
+    _marker: PhantomData<&'a ()>
+}
+impl <'a> JsonValueFormatter<'a> {
+    pub fn value_formatter() -> ValueFormatter<'a, Self> {
+        ValueFormatter::default()
+    }
+}
+
+impl <'a> ValueFormatterVisitor<'a> for JsonValueFormatter<'a> {
+    fn visit_null(_: &ValueFormatter<'a, Self>) -> Cow<'a, str> {
+        Cow::Borrowed("null")
+    }
+
+    fn visit_bool(_: &ValueFormatter<'a, Self>, v: &'a bool) -> Cow<'a, str> {
+        if *v {
+            Cow::Borrowed("true")
+        }
+        else {
+            Cow::Borrowed("false")
+        }
+    }
+
+    fn visit_i64(_: &ValueFormatter<'a, Self>, v: &'a i64) -> Cow<'a, str> {
+        Cow::Owned(v.to_string())
+    }
+
+    fn visit_u64(_: &ValueFormatter<'a, Self>, v: &'a u64) -> Cow<'a, str> {
+        Cow::Owned(v.to_string())
+    }
+
+    fn visit_f64(_: &ValueFormatter<'a, Self>, v: &'a f64) -> Cow<'a, str> {
+        Cow::Owned(v.to_string())
+    }
+
+    fn visit_str(_: &ValueFormatter<'a, Self>, v: &'a str) -> Cow<'a, str> {
+        let bytes = v.as_bytes();
+        if v.len() > 0 && bytes[0] == b'"' {
```

Comment:
Fair point, I might remove that sanitisation for strings and open a new issue for it. That way if you're happy with the rest of the design in this PR maybe we can push it through and cut a release after it sanitises properly.

I still need to do object serialisation, but have been procrastinating it.